pkg/server/test/connected_objects.go

```diff
@@ -1216,15 +1235,165 @@ func ConnectedObjectsTest(t *testing.T, ds storage.OpenFGADatastore) {
+				},
+				{
+					Object:       "folder:folder2",
+					ResultStatus: connectedobjects.NoFurtherEvalStatus,
+				},
+				{
+					Object:       "folder:folder3",
+					ResultStatus: connectedobjects.NoFurtherEvalStatus,
+				},
+			},
+		},
+		{
+			name: "respects_max_results",
+			request: &connectedobjects.ConnectedObjectsRequest{
+				StoreID:    ulid.Make().String(),
+				ObjectType: "folder",
+				Relation:   "viewer",
+				User: &connectedobjects.UserRefObject{
+					Object: &openfgav1.Object{
+						Type: "user",
+						Id:   "jon",
+					},
+				},
+				ContextualTuples: []*openfgav1.TupleKey{
+					tuple.NewTupleKey("folder:folder3", "viewer", "user:jon"),
+				},
+			},
+			minResultsExpected: 2,
+			maxResults:         2,
+			model: `
+			type user
+
+			type folder
+			  relations
+			    define viewer: [user] as self
+			`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:folder2", "viewer", "user:jon"),
+			},
+			allResults: []*connectedobjects.ConnectedObjectsResult{
+				{
+					Object:       "folder:folder1",
+					ResultStatus: connectedobjects.NoFurtherEvalStatus,
+				},
+				{
+					Object:       "folder:folder2",
+					ResultStatus: connectedobjects.NoFurtherEvalStatus,
+				},
+				{
+					Object:       "folder:folder3",
+					ResultStatus: connectedobjects.NoFurtherEvalStatus,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := context.Background()
+			store := ulid.Make().String()
+			test.request.StoreID = store
+
+			model := &openfgav1.AuthorizationModel{
+				Id:              ulid.Make().String(),
+				SchemaVersion:   typesystem.SchemaVersion1_1,
+				TypeDefinitions: parser.MustParse(test.model),
+			}
+			err := ds.WriteAuthorizationModel(ctx, store, model)
+			require.NoError(t, err)
+
+			err = ds.Write(ctx, store, nil, test.tuples)
+			require.NoError(t, err)
+
+			connectedObjectsCmd := connectedobjects.NewConnectedObjectsQuery(ds, typesystem.New(model),
+				connectedobjects.WithMaxResults(test.maxResults))
+
+			resultChan := make(chan *connectedobjects.ConnectedObjectsResult, 100)
+			done := make(chan struct{})
+
+			var results []*connectedobjects.ConnectedObjectsResult
+			go func() {
+				for result := range resultChan {
+					results = append(results, result)
+				}
+
+				done <- struct{}{}
+			}()
+
+			timeoutCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+
+			go func() {
+				err = connectedObjectsCmd.Execute(timeoutCtx, test.request, resultChan)
+				require.NoError(t, err)
+				close(resultChan)
+			}()
+
+			select {
+			case <-timeoutCtx.Done():
+				require.FailNow(t, "timed out waiting for response")
+			case <-done:
+			}
 
-				require.Equal(test.expectedResult, results)
+			require.GreaterOrEqual(t, len(results), int(test.minResultsExpected))
+			if test.maxResults > 0 {
```

Comment:
should the condition be 
```suggestion
			if test.maxResults == test.minResultsExpected {
```

In this way, we only check for subset if we know that the results may be cut short due to max results limitation.

Reply:
This comment doesn't apply anymore. ConnectedObjects now doesn't accept a `maxResults`